source/slow-down.ts

```diff
@@ -0,0 +1,202 @@
+// /source/express-slow-down.ts
+// The speed limiting middleware.
+
+import type { Request, Response, NextFunction } from 'express'
+import { rateLimit, type Options as RateLimitOptions } from 'express-rate-limit'
+import type {
+	SlowDownRequestHandler,
+	AugmentedRequest,
+	SlowDownOptions,
+	Options,
+} from './types'
+
+/**
+ * Remove any options where their value is set to undefined. This avoids overwriting defaults
+ * in the case a user passes undefined instead of simply omitting the key.
+ *
+ * @param passedOptions {Options} - The options to omit.
+ *
+ * @returns {Options} - The same options, but with all undefined fields omitted.
+ *
+ * @private
+ */
+const omitUndefinedOptions = (
+	passedOptions: Partial<Options>,
+): Partial<Options> => {
+	const omittedOptions: Partial<Options> = {}
```

Comment:
`filteredOptions` might be a better name for this variable; `omittedOptions` suggests to me that it's the opposite of what it actually is.